### PR TITLE
1 - Add Trending to report

### LINKED ISSUE #188: Empty Charts Issue
# 🐛 Summary #

Charts in the reports that have no data show negative numbers on the Y Axis.  It would be preferable to show all charts starting at 0 as we are only dealing with positive numbers in general.

## To reproduce ##

1. Review any report that has no data, and you will see the chart's Y axis in the negative.

## Expected behavior ##

Charts start at zero and count up.

```

### PR DESCRIPTION
**Pull request by DJensen94**
Fixes #188 
Add Trending to the credentials and dark web pages of the report
## 🗣 Description ##
This PR adds Trending to two pages in the report to show how many cred breaches/mentions occured compared to the previous weeks
## 💭 Motivation and context ##
We want to give our stakeholders context for the current weeks numbers, seeing how these results compare to past weeks can help them see fluctuations more easily

## ✅ Pre-approval checklist ##

- [X] This PR has an informative and human-readable title.
- [X] Changes are limited to a single goal - *eschew scope creep!*
- [x] All new and existing tests pass.

### CODE REVIEW

**Review by aloftus23 — APPROVED** [review]
Looks good to me.

**Comment by aloftus23** [comment]
@dav3r The PRs are starting to stack up so we are numbering them by priority in the title. This is number 1 when you get the chance to review.

**Review by cduhn17 — APPROVED** [review]
Looks good.

**Comment by cduhn17** [comment]
@dav3r, 
Do you have a time frame of when you might get a chance to take a look at this PR? Thanks.

**Comment by dav3r** [comment]
> @dav3r, Do you have a time frame of when you might get a chance to take a look at this PR? Thanks.

I'm going to take a look at it now.

File: src/pe_reports/charts.py
Comment (dav3r) [inline src/pe_reports/charts.py:93]: In general, it would be nicer to see smaller, more targeted commits.  That way changes, like all of those to `charts.py` can be documented in the commit log, rather than being hidden amongst a larger commit like https://github.com/cisagov/pe-reports/pull/192/commits/311223a8034efe87cca37fe8e8183daa522b0918.

Some more comments sprinkled in with the code would also greatly improve its readability.

File: src/pe_reports/charts.py
Comment (dav3r) [inline src/pe_reports/charts.py]: Generally, we recommend removing commented-out/unused code.  If there is a valid reason to keep it around, please add a comment explaining why.

File: src/pe_reports/charts.py
Comment (dav3r) [inline src/pe_reports/charts.py]: See comment above about removing unused code.

File: src/pe_reports/charts.py
Comment (dav3r) [inline src/pe_reports/charts.py]: There is a bunch of duplicated code here.  Are the only values for `len(df.columns)` 1 or 2?  If so, you could add another outer loop like this:
```suggestion
        for col in range(len(df.columns)):
            for i, j in df[df.columns[col]].items():
                if int(j):
                    plt.annotate(
                        str(int(j)),
                        xy=(i, j),
                        textcoords="offset points",  # how to position the text
                        xytext=(
                            0,
                            5,
                        ),  # distance from text to points (x,y)
                        ha="center",  # horizontal alignment can be left, right or center
                    )
```

Note that I removed the commented-out `fontsize=2` line above.

File: src/pe_reports/data/db_query.py
Comment (dav3r) [inline src/pe_reports/data/db_query.py]: Remove errant extra space:
```suggestion
    """Query credentials view."""
```

File: src/pe_reports/metrics.py
Comment (dav3r) [inline src/pe_reports/metrics.py]: Might as well just consolidate these lines:
```suggestion
        return len(self.creds_view[self.creds_view["password_included"]])
```

File: src/pe_reports/metrics.py
Comment (dav3r) [inline src/pe_reports/metrics.py]: Same here:
```suggestion
        return self.creds_view.shape[0]
```

File: src/pe_reports/pages.py
Comment (dav3r) [inline src/pe_reports/pages.py:275]: You should add a comment explaining why you are using this 27 day magic number.

File: tests/test_pe_reports.py
Comment (dav3r) [inline tests/test_pe_reports.py]: First of all, I applaud you for adding a new test here! 👍   However, I'm not so sure that you are really testing your `get_orgs` query.  If I'm understanding this correctly (which I may not be), you are mocking up the DB connection and telling it what you want the results to be (`query_result`), then you are validating that the query results are the same as what you said you wanted the results to be- which they always should be.  Am I missing something or is the `get_orgs` query not really being tested here?

**Review by dav3r — CHANGES_REQUESTED** [review]
Take a look at my feedback and please let me know if you have any questions.

File: src/pe_reports/charts.py
Comment (DJensen94) [reply src/pe_reports/charts.py:93]: @dav3r  resolved in commit [453382d](https://github.com/cisagov/pe-reports/pull/192/commits/453382d3e42200b666790f6dd07cf010b8a58c27)

File: src/pe_reports/charts.py
Comment (DJensen94) [reply src/pe_reports/charts.py]: @dav3r issue resolved in commit [453382d](https://github.com/cisagov/pe-reports/pull/192/commits/453382d3e42200b666790f6dd07cf010b8a58c27)

File: src/pe_reports/charts.py
Comment (DJensen94) [reply src/pe_reports/charts.py]: @dav3r issue was resolved in [453382d](https://github.com/cisagov/pe-reports/pull/192/commits/453382d3e42200b666790f6dd07cf010b8a58c27)

File: src/pe_reports/pages.py
Comment (DJensen94) [reply src/pe_reports/pages.py:275]: @dav3r resolved in commit [e0a3626](https://github.com/cisagov/pe-reports/pull/192/commits/e0a3626a892cd8b547b59f9ae0aef5a13e503765)

File: tests/test_pe_reports.py
Comment (DJensen94) [reply tests/test_pe_reports.py]: @dav3r removed this test in commit [b6288ae](https://github.com/cisagov/pe-reports/pull/192/commits/b6288aeebe7dcb344bb564973970e586e1988fa3)

Comment (aloftus23) [reply tests/test_pe_reports.py]: Decided to create a new issue to write more useful tests: #198

Comment (dav3r) [reply tests/test_pe_reports.py]: Thanks @DJensen94 and @aloftus23 👍

File: src/pe_reports/charts.py
Comment (dav3r) [inline src/pe_reports/charts.py]: Typo:
```suggestion
        # Specify axis attributes
```

**Review by dav3r — APPROVED** [review]
Once the typo mentioned [here](https://github.com/cisagov/pe-reports/pull/192#discussion_r868409603) is corrected, this one looks good to me. 👍

**Review by jsf9k — CHANGES_REQUESTED** [review]
I'd like to see [the two warnings from LGTM](https://github.com/cisagov/pe-reports/pull/192#issuecomment-1121421689) resolved before merging.

**Comment by DJensen94** [comment]
> I'd like to see [the two warnings from LGTM](https://github.com/cisagov/pe-reports/pull/192#issuecomment-1121421689) resolved before merging.

@dav3r , @jsf9k We are pretty positive that the warnings are unfounded, they claim that origin is not a parameter for 
grouper, but it is a new parameter:
image
So LGTM is out of date

**Comment by jsf9k** [comment]
> > I'd like to see [the two warnings from LGTM](https://github.com/cisagov/pe-reports/pull/192#issuecomment-1121421689) resolved before merging.
> 
> @dav3r , @jsf9k We are pretty positive that the warnings are unfounded, they claim that origin is not a parameter for grouper, but it is a new parameter: image So LGTM is out of date

Then you should add [a suppression comment](https://lgtm.com/help/lgtm/alert-suppression#python) for those two warnings.

**Comment by DJensen94** [comment]
> > > I'd like to see [the two warnings from LGTM](https://github.com/cisagov/pe-reports/pull/192#issuecomment-1121421689) resolved before merging.
> > 
> > 
> > @dav3r , @jsf9k We are pretty positive that the warnings are unfounded, they claim that origin is not a parameter for grouper, but it is a new parameter: image So LGTM is out of date
> 
> Then you should add [a suppression comment](https://lgtm.com/help/lgtm/alert-suppression#python) for those two warnings.

@jsf9k we have added the suppression comment. LGTM says the warning won't be removed till after the merge occurs.

**Review by jsf9k — APPROVED** [review]
I can't get the LGTM thing to go away either, so I'll have to let it pass for now.

**Pull request merged**
